lib/interactive_viewer.ml

```diff
@@ -133,4 +142,37 @@ let view (patches : Patch.t list) =
                the previous operation");
     ]
 
-let start patch = Ui_loop.run ~quit ~tick_period:0.2 (view patch)
+let start patch events test =
+  if test then (
+    let z_patches : 'a Zipper.t Lwd.var =
+      match Zipper.zipper_of_list patch with
+      | Some z -> Lwd.var z
+      | None -> failwith "zipper_of_list: empty list"
+    in
+    let handle_event (event : Notty.Unescape.event) : Ui.may_handle =
```

Comment:
It's counter-productive to repeat the key handler, instead we should use `(view patch)` without changing it.

Looking at Nottui's documentation, I found that `Renderer.dispatch_key renderer event` can send a key event to the UI. You already have a renderer created by `content_image_renderer` so this should be easy to use.

Reply:
@Julow , just adding this here for reference and clarity: 
```ocaml
let content_ui = Lwd.quick_sample (Lwd.observe (view patch)) in    
let ui_renderer = content_image_renderer content_ui in

let ui_events = (List.map (fun e -> convert_char_to_key e) events) in

List.iter ( function | event ->
begin
  let dispatch_res = Renderer.dispatch_key !ui_renderer event in
  
  match dispatch_res with
  | `Handled -> print_endline "Handled event";
  | `Unhandled -> print_endline "Unhandled event";

  ignore (dispatch_res)
end
```

I initially looked through the docs and how to use `Renderer.dispatch_key`. I must admit that it is a much more feasible and elegant :smile: option but while testing, I noticed that the events were not getting processed as expected. On printing out if the event was handled, I always get "Unhandled event" response. I have looked through the implementation of `Renderer.dispatch_key` and I am sure this is due to the ui not having a focus for the event - or the event listener created by `Ui.keyboard_area` not getting the event. I have tried several methods to set focus but I am not quite sure on how to go about it after several failed attempts. Please can I have some hint on this?

Reply:
It's normal to have some events that are unhandled (the ui do not handle all the keys) but this code don't seem to contain an obvious bug. Could you push your latest change to the branch so I can have a look ?

Reply:
I have pushed my latest changes to this branch